GrandCentralBoard/Widgets/Image/Source/RemoteImageSource.swift

```diff
@@ -24,30 +23,49 @@ enum RemoteImageSourceError : ErrorType, HavingMessage {
     }
 }
 
+struct Counter {
+    private(set) var value = 0
+    let modulo: Int
+
+    mutating func nextValue() -> Int {
+        let returnValue = value
+        value = (value + 1) % modulo
+        return returnValue
+    }
+}
+
 final class RemoteImageSource : Asynchronous {
 
     typealias ResultType = GrandCentralBoardCore.Result<Image>
 
     let interval: NSTimeInterval
     let sourceType: SourceType = .Momentary
+    let dataDownloader: DataDownloader
 
-    private let url: NSURL
+    private let paths: [String]
+    private var counter: Counter
 
-    init(url: NSURL, interval: NSTimeInterval = 30) {
+    init(paths: [String], dataDownloader: DataDownloader, interval: NSTimeInterval = 30) {
         self.interval = interval
-        self.url = url
+        self.paths = paths
+        self.dataDownloader = dataDownloader
+        self.counter = Counter(value: 0, modulo: paths.count)
     }
 
     func read(closure: (ResultType) -> Void) {
-        Alamofire.request(.GET, url).response { (request, response, data, error) in
-
-            if let data = data, image = UIImage(data: data) {
-                let image = Image(value: image, time: NSDate())
-                closure(.Success(image))
-                return
+        let path = paths[counter.nextValue()]
+        dataDownloader.downloadDataAtPath(path) { result in
+            switch result {
+            case .Success(let data):
+                if let value = UIImage(data: data) {
+                    let image = Image(value: value, time: NSDate())
```

Comment:
Why do we need to wrap the image? The method's caller only uses the UIImage value anyway.